kratos/utilities/assign_master_slave_constraints_to_neighbours_utility.cpp

```diff
@@ -0,0 +1,247 @@
+    KRATOS_CATCH(""); 
+} 
+
+// Search for neighboring nodes (in rMasterStructureNodes) of the given slave node
+void AssignMasterSlaveConstraintsToNeighboursUtility::SearchNodesInRadiusForNode(
+    NodeType::Pointer pSlaveNode,
+    double const Radius,
+    ResultNodesContainerType& rCloudOfNodes)
+{
+
+    KRATOS_TRY;
+
+    ResultNodesContainerType local_results(mMaxNumberOfNodes);
+    std::size_t num_of_results = 0;
+    auto results_iterator = local_results.begin();
+    num_of_results = mpBins->SearchInRadius(*pSlaveNode, Radius, results_iterator, mMaxNumberOfNodes);
+    rCloudOfNodes.insert(rCloudOfNodes.begin(), local_results.begin(), local_results.begin() + num_of_results);
+
+    KRATOS_CATCH("");
+}
+
+void AssignMasterSlaveConstraintsToNeighboursUtility::GetDofsAndCoordinatesForSlaveNode(
+    NodeType::Pointer pSlaveNode,
+    const std::vector<std::reference_wrapper<const Kratos::Variable<double>>>& rVariableList,
+    std::vector<DofPointerVectorType>& rSlaveDofs,
+    array_1d<double, 3>& rSlaveCoordinates
+)
+{
+    KRATOS_TRY;
+
+    // Check if the node has the required DOFs for all variables in the list
+    bool has_all_dofs = true;
+    for (const auto& variable : rVariableList) {
+        if (!pSlaveNode->HasDofFor(variable.get())) {
+            has_all_dofs = false;
+            break;
+        }
+    }
+
+    if (has_all_dofs) {
+        // Get DOFs for each variable in the list
+        rSlaveDofs.resize(rVariableList.size());
+        for (std::size_t i = 0; i < rVariableList.size(); ++i) {
+            const auto& variable = rVariableList[i];
+            rSlaveDofs[i].resize(1);
+            rSlaveDofs[i][0] = pSlaveNode->pGetDof(variable.get());
+        }
+    } else {
+        std::stringstream variables_str;
+        variables_str << "[";
+        for (const auto& variable : rVariableList) {
+            variables_str << variable.get() << ", ";
+        }
+        variables_str.seekp(-2, std::ios_base::end); // Remove the last ", "
+        variables_str << "]";
+        KRATOS_ERROR << "The node with ID " << pSlaveNode->Id() << " does not have the required DOFs for one of the variables " << variables_str.str() << std::endl;
+    }
+
+    rSlaveCoordinates = pSlaveNode->Coordinates();
+
+    KRATOS_CATCH("");
+}
+
+// Get DOFs and Coordinates arrays for a given variable list for a cloud of nodes
+void AssignMasterSlaveConstraintsToNeighboursUtility::GetDofsAndCoordinatesForCloudOfNodes(
+    const ResultNodesContainerType& CloudOfNodesArray,
+    const std::vector<std::reference_wrapper<const Kratos::Variable<double>>>& rVariableList,
+    std::vector<DofPointerVectorType>& rCloudOfDofs,
+    Matrix& rCloudOfNodesCoordinates
+)
+{
+    KRATOS_TRY;
+
+    // Resize rCloudOfDofs to match the size of the variable list
+    rCloudOfDofs.resize(rVariableList.size());
+
+    // Resize rCloudOfDofs vectors to match the size of the cloud of nodes
+    for (auto& dofs : rCloudOfDofs)
```

Comment:
Usually we consider https://en.wikipedia.org/wiki/Indentation_style#K&R_style with Linux kernel variant

Reply:
At least I consider that and I ask other to use that

Reply:
Will update this.